iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m

```diff
@@ -202,32 +211,39 @@ - (BOOL)isRootViewAttached
 }
 
 #pragma mark - TiWindowProtocol Base Methods
-- (void)open:(id)args
+- (KrollPromise *)open:(id)args
 {
-  //If an error is up, Go away
+  JSContext *context = [self currentContext];
+
+  // If an error is up, Go away
   if ([[[[TiApp app] controller] topPresentedController] isKindOfClass:[TiErrorNavigationController class]]) {
     DebugLog(@"[ERROR] ErrorController is up. ABORTING open");
-    return;
+    return [KrollPromise rejectedWithErrorMessage:@"ErrorController is up. ABORTING open" inContext:context];
   }
 
-  //I am already open or will be soon. Go Away
+  // I am already open or will be soon. Go Away
   if (opening || opened) {
-    return;
+    return [KrollPromise rejectedWithErrorMessage:@"Window is already opened or opening." inContext:context];
   }
 
-  //Lets keep ourselves safe
+  // Lets keep ourselves safe
   [self rememberSelf];
 
+  // Don't recreate if we're calling ourselves again because root view is not loaded/attached!
+  if (openPromise == nil) {
+    openPromise = [[[KrollPromise alloc] initInContext:context] retain];
```

Comment:
```suggestion
    openPromise = [[KrollPromise alloc] initInContext:context];
```
No need of retain.  For safer side openPromise should be release inside -(void)dealloc as well.